File: ICMS.Entity/Entities/Activity.cs
using System;
using System.Collections.Generic;

namespace ICMS.Entity.Entities
{
    public class Activity
    {
        public string ActivityId { get; set; }
        public string StudentId { get; set; }
        public string ActivityTypeCode { get; set; }
        public string ActivityTypeDescription { get; set; }
        public string Description { get; set; }
        public string PositionTitle { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public string OrganisationId { get; set; }
        public string OrganisationName { get; set; }
        public string CountryCode { get; set; }
        public string CountryName { get; set; }
        public decimal Duration { get; set; }
        public decimal? UnitRecognised { get; set; }
        public decimal ProBonoRecognised { get; set; }
        public string OverallGradeCode { get; set; }
        public string OverallGradeDescription { get; set; }
        public string StatusCode { get; set; }
        public string StatusDescription { get; set; }
        public string Unit { get; set; }
        public string OverallSummaryStatus { get; set; }
        public string ProgrammeCode { get; set; }
        public string Url { get; set; }

        public List<KeyValuePair<string, decimal?>> UnitsRecognised { get; set; }
        public List<LearningObjective> LearningObjectives { get; set; }
        public List<ActivityQuestion> ActivityQuestions { get; set; }
    }
}

File: ICMS/src/app/activity/activity.ts

export class Activity {
    ActivityId: string;
    Description: string;
    ActivityTypeCode: string;
    ActivityTypeDescription: string;
    ProgrammeDescription: string;
    PositionTitle: string;
    StartDate: string;
    EndDate: string;
    OrganisationDescription: string;
    CountryDescription: string;
    Duration: number;
    UnitRecognised: number;
    OverallGradeDescription: string;
    ReflectionStatusCode: string;
    ReflectionStatusDescription: string;
    Unit: string;
    ShowPreActivity: boolean;
    ShowMidActivity: boolean;
    ShowPostActivity: boolean;
    EnablePreActivity: boolean;
    EnableMidActivity: boolean;
    EnablePostActivity: boolean;
    UnitsRecognised: any[];
    Url: string;
}

export class OverallSummary {
    Description: string;
    Status: string;
    UnitsRequired: number;
    UnitsRecognized: number;
    ProBono: number;
    UnitsRemaining: number;
    Exempted: number;
    ProgrammeCode: string;
}

export class UpdateActivity {
    ActivityId: string;
    LearningObjectives?: PostLearningObjective[];
    ActivityQuestionAnswers?: PostActivityQuestion[];
    Stage: string;
}

export class PostLearningObjective {
    IsSelected?: boolean;
    IsMidActivityOnTarget?: boolean;
    IsObjectiveAchieved?: boolean;
    LearningObjectiveId: string;
    LearningObjectiveQuestions: PostLearningObjectiveQuestion[];
}

export class PostLearningObjectiveQuestion {
    LearningObjectiveQuestionId: string;
    Answer: string;
}

export class PostActivityQuestion {
    ActivityQuestionId: string;
    Answer: string;
}

export class UpdateActivityStatus {
    StudentId: string;
    ActivityId: string;
    Status: string;
}